sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -106,9 +110,15 @@ public ClientRetryPolicy(GlobalEndpointManager globalEndpointManager,
                 } else {
                     return this.shouldNotRetryOnEndpointFailureAsync(this.isReadRequest, false);
                 }
+            } else if (clientException != null && clientException.getCause() instanceof ReadTimeoutException) {
```

Comment:
NIT: similar as above - have a helper function IsNonIOBoundOperation or similar that returns true for now if addres refresh or query plan retrieval but can be extended easily

Reply:
We already have a utility class/methods for similar functionality. 
This should go to the same class as `WebExceptionUtility.isNetworkFailure()`.

Reply:
This done, we added the retry utility.